services/antivirus/pkg/service/service.go

```diff
@@ -116,21 +117,27 @@ func (av Antivirus) Run() error {
 		return err
 	}
 
-	for e := range ch {
-		err := av.processEvent(e, natsStream)
-		if err != nil {
-			switch {
-			case errors.Is(err, ErrFatal):
-				return err
-			case errors.Is(err, ErrEvent):
-				// Right now logging of these happens in the processEvent method, might be cleaner to do it here.
-				continue
-			default:
-				av.l.Fatal().Err(err).Msg("unknown error - exiting")
+	wg := sync.WaitGroup{}
+	for i := 0; i < av.c.Workers; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for e := range ch {
```

Comment:
This loop never be broken because we are not closing a channel `ch`

Reply:
It could be like this, but graceful shutdown anyway not working
```
	wg := sync.WaitGroup{}
	for i := 0; i < av.c.Workers; i++ {
		wg.Add(1)
		go func() {
			defer wg.Done()
			for {
				select {
				case e, ok := <-ch:
					if !ok {
						return
					}
					err := av.processEvent(e, natsStream)
					if err != nil {
						switch {
						case errors.Is(err, ErrFatal):
							av.l.Fatal().Err(err).Msg("fatal error - exiting")
						case errors.Is(err, ErrEvent):
							av.l.Error().Err(err).Msg("continuing")
						default:
							av.l.Fatal().Err(err).Msg("unknown error - exiting")
						}
					}
				case <-av.c.Context.Done():
					av.l.Debug().Msg("antivirus worker stopped")
					return
				}
			}
		}()
	}
	wg.Wait()
```

Reply:
Let's go for it as is. We can later rework for graceful shutdowns.